### DISCUSSION TITLE
Remix Splat Routes

### DISCUSSION THREAD

**Discussion by InceptionCode** [sandulat/routes-gen#35] [Q&A]
I have a splat route that is not correctly generating as a route type. Is anyone else having the same issue? How can you work with splat routes?

### Splat Routes

[splats documentation](https://remix.run/docs/en/main/guides/routing#md-splats)

**Comment by sandulat** [comment]
@InceptionCode Could you please share a screenshot of your routes and the generated type with the issue?

**Reply by InceptionCode** [reply]
<img width="1780" alt="Screenshot 2023-03-16 at 12 49 04 PM" src="https://user-images.githubusercontent.com/16050250/225693935-d98d3d2c-ad2c-4b97-b743-14117d4e10ae.png">

The route in question is `/browse/playlists`. This route is defined as a splat route that should match everything after `playlists/`. So I expected the generated type to have a wildcard, `RouteParam` .

<img width="1426" alt="Screenshot 2023-03-16 at 12 51 41 PM" src="https://user-images.githubusercontent.com/16050250/225694467-fdd736e8-09d0-4813-ac14-2c7b205353c5.png">

Instead, `RouteParms` is defined as `Record<string, never>;`

<img width="1426" alt="Screenshot 2023-03-16 at 12 50 37 PM" src="https://user-images.githubusercontent.com/16050250/225694745-24b7898d-f2ee-48c1-a5ae-6622912347a4.png">

I don't know what the best way to handle this is off the top of my head. Considering a splat route doesn't include a named parameter. @sandulat

**Reply by sandulat** [reply]
@InceptionCode A splat route can accept anything, so there is no way to add typings for that.
Why not just use `route("/browser/playlists") + "/whatever/foo/bar"`?

**Reply by InceptionCode** [reply]
I honestly did not think about that at all. That definitely works! 🎉 @sandulat
